src/python/pants/jvm/resolve/lockfile_metadata.py

```diff
@@ -57,15 +54,6 @@ def from_lockfile(
             ),
         )
 
-    def _header_dict(self) -> dict[Any, Any]:
```

Comment:
Please don't delete this, it's defensive to ensure that future implementers don't override this incorrectly (primarily by making sure the instructions in the docstring are preserved in every file where `LockfileMetadata` is extended).

This is definitely a sign that the API needs refactoring (and I'll take that on), but in the meantime, this method should remain to prevent future foot-shooting while the current structure is in place.

Reply:
Okay

Reply:
Wait actually I don't think I get the motivation. You'll have the docstring from `core/util_rules/lockfile_metadata.py` already. Plus the concrete examples from both JVM and Python for what to do.

This seems like a violation of DRY, and I don't understand what keeping these lines is worth the noise

Reply:
I preface this by saying that this code structure (which I wrote) is really bad, because it's impossible to validate that it's been used correctly, and I absolutely need to re-write it. Reasoning about it like it's good code is a bad idea.

The suggestion I'm making is _absolutely_ a violation of DRY. The problem is that forgetting to call `super()` in this case creates an error because of how the versioning is implemented in the root class, and it's an error that I found difficult to test for back when I wrote the Python version.

Basically the suggestion, it's a defensive mechanism, to make sure the warning is front and centre to everyone who's editing the file, because it's easy to get it wrong. I think it's necessary because I wrote it and _I_ found it difficult to work with.

My aim here is to give the next person to touch the code (presuming I don't fix the structure) every chance to succeed when they touch it next. It leaves a bad taste in my mouth, but that's how it is for the moment.

Reply:
(I've added the refactor as a TODO for myself at #14219; I have an idea of what a better pattern would look like)